src/main/java/seedu/duke/command/EditCommand.java

```diff
@@ -45,29 +105,22 @@ public EditCommand(String input) {
      * @return an array of Strings containing the extracted values of the fields to be updated.
      */
     public String[] extract(String input) {
-        String[] extractedFields = new String[4];
+        String[] extractedFields = new String[3];
 
         // Define regular expression patterns
-        Pattern namePattern = Pattern.compile("p/([\\w\\s]+)");
-        Pattern newFolderPattern = Pattern.compile("-f nf/([\\w\\s]+)");
-        Pattern newDescriptionPattern = Pattern.compile("-d nd/([\\w\\s]+)");
-        Pattern newNamePattern = Pattern.compile("-N np/([\\w\\s]+)");
+        final Pattern namePattern = Pattern.compile("p/([\\w\\s]+)");
```

Comment:
This assumes that all 3 must be provided. Is this helpful? If someone wants to edit everything they can jsut delete and create a new one no?